packages/shared/lib/ledger.ts

```diff
@@ -6,62 +6,91 @@ import { api } from 'shared/lib/wallet'
 import { get, writable } from 'svelte/store'
 import { localize } from './i18n'
 
-export const ledgerSimulator = true
+export const ledgerSimulator = false
 export const isLedgerConnected = writable<boolean>(true)
 export const isLedgerLegacyConnected = writable<boolean>(true)
 
-const DEFAULT_LEDGER_STATUS_POLL_INTERVAL = 1000
+const LEDGER_STATUS_POLL_INTERVAL_ON_DISCONNECT = 1500
 
-export function getLedgerDeviceStatus(): void {
+let polling = false
+let intervalTimer
+
+export function getLedgerDeviceStatus(onConnected = () => { }, onDisconnected = () => { }, onError = () => { }) {
     api.getLedgerDeviceStatus(ledgerSimulator, {
         onSuccess(response) {
             let _isLedgerConnected = response.payload?.type === LedgerStatus.Connected
+            isLedgerConnected.set(_isLedgerConnected)
             if (_isLedgerConnected) {
-                closePopup()
+                onConnected()
             } else {
-                openLedgerNotConnectedPopup()
+                onDisconnected()
             }
-            isLedgerConnected.set(_isLedgerConnected)
-        },
-        onError() {
-            openLedgerNotConnectedPopup()
         },
+        onError(err) {
+            onError()
+        }
     })
 }
 
-function openLedgerNotConnectedPopup(legacy: boolean = false) {
+export function promptUserToConnectLedger(
+    onConnected = () => { },
+    onCancel = () => { },
+) {
+    const _onCancel = () => {
+        stopPollingLedgerStatus()
+        onCancel()
+    }
+    const _onConnected = () => {
+        stopPollingLedgerStatus()
+        if (get(popupState).active) {
+            closePopup()
+        }
+        onConnected()
+    }
+    const _onDisconnected = () => {
+        pollLedgerDeviceStatus(LEDGER_STATUS_POLL_INTERVAL_ON_DISCONNECT, _onConnected, _onDisconnected, _onCancel)
+        if (!get(popupState).active) {
+            openLedgerNotConnectedPopup(false, onCancel)
+        }
+    }
+    getLedgerDeviceStatus(_onConnected, _onDisconnected, _onCancel)
+}
+
+export function pollLedgerDeviceStatus(pollInterval, _onConnected = () => { }, _onDisconnected = () => { }, _onCancel = () => { }) {
+    if (!polling) {
+        intervalTimer = setInterval(async () => {
+            getLedgerDeviceStatus(_onConnected, _onDisconnected, _onCancel)
```

Comment:
If we also call `getLedgerDeviceStatus(_onConnected, _onDisconnected, _onCancel)` right before starting the internal we can get the ledger status right at subscription time